### PR TITLE
Allow to move a fatman from one infrastructure target to another

### PR DESCRIPTION
**Pull request by iszulcdeepsense**

### CODE REVIEW

File: lifecycle/lifecycle/deployer/redeploy.py [hunk 1]
```diff
@@ -83,6 +83,51 @@ def reprovision_fatman(
...
+    deployer_username: str,
+    auth_subject: Optional[models.AuthSubject],
+):
+    """Move fatman from one infrastructure target to another"""
+    fatman = read_fatman(fatman_name, fatman_version, config)
+    manifest = fatman.manifest
+    old_infra_target = fatman.infrastructure_target
+    assert fatman.manifest is not None, "fatman doesn't have Manifest data specified"
+    assert fatman.image_tag is not None, "fatman's image tag is unknown"
+    assert new_infra_target, "infrastructure target has to be specified"
```
Comment (LookACastle) [inline lifecycle/lifecycle/deployer/redeploy.py:101]:
```suggestion
    assert new_infra_target is not None, "infrastructure target has to be specified"
```

I assume you want to assert some property (likely that it's not None) about `new_infra_target`?

File: lifecycle/lifecycle/deployer/redeploy.py [hunk 2]
```diff
@@ -83,6 +83,51 @@ def reprovision_fatman(
...
+        if manifest.secret_runtime_env_file:
+            fatman_secrets = _retrieve_fatman_secrets(config, manifest, plugin_engine, fatman)
+        else:
+            fatman_secrets = FatmanSecrets(git_credentials=None, secret_build_env={}, secret_runtime_env={})
+
+        with wrap_context('deploying fatman to a new infrastructure'):
+            provision_fatman(
+                config, manifest, fatman.image_tag, fatman_secrets.secret_build_env,
+                fatman_secrets.secret_runtime_env, deployment,
+                auth_subject, None, plugin_engine,
```
Comment (LookACastle) [inline lifecycle/lifecycle/deployer/redeploy.py:115]:
```suggestion
                auth_subject, None, plugin_engine
```

Nitpick: Trailing comma

File: racetrack_client/racetrack_client/main.py [hunk 3]
```diff
@@ -69,10 +70,17 @@ def _print_help(_: argparse.Namespace):
...
     parser_delete.add_argument('workdir', default='.', nargs='?', help='directory with fatman.yaml manifest')
     parser_delete.add_argument('racetrack_url', default='', nargs='?', help='URL to Racetrack server or alias name')
-    parser_delete.add_argument('--version', nargs='?', type=str,
-                               help='fatman version to delete')
+    parser_delete.add_argument('--version', nargs='?', type=str, help='fatman version to delete')
     parser_delete.set_defaults(func=_delete_fatman)
 
+    # racetrack move
+    parser_move = subparsers.add_parser('move', help='Move fatman from one infrastructure target to another fatman instance')
+    parser_move.add_argument('--remote', type=str, help='URL to Racetrack server or alias name')
```
Comment (LookACastle) [inline racetrack_client/racetrack_client/main.py:78]:
```suggestion
    parser_move.add_argument('--remote', type=str, help='Racetrack server\'s URL or alias name')
```

Nitpick: this feels like more "natural" English to me.

File: racetrack_client/racetrack_client/main.py [hunk 4]
```diff
@@ -69,10 +70,17 @@ def _print_help(_: argparse.Namespace):
...
+    parser_move.add_argument('--name', required=True, type=str, help='fatman name to move out')
+    parser_move.add_argument('--version', required=True, type=str, help='fatman version to move out')
+    parser_move.add_argument('--infrastructure', required=True, type=str, help='infrastructure target to move on')
```
Comment (LookACastle) [inline racetrack_client/racetrack_client/main.py:81]:
```suggestion
    parser_move.add_argument('--infrastructure', required=True, type=str, help='infrastructure target to move to')
```

Nitpick: this feels like more "natural" English to me.

**Review by LookACastle — CHANGES_REQUESTED** [review]

File: lifecycle/lifecycle/deployer/redeploy.py (hunk 1 above)
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/deployer/redeploy.py:101]: `assert new_infra_target` is stronger as it checks if it's neither None nor empty. See https://stackoverflow.com/a/9573283/6772197

File: racetrack_client/racetrack_client/main.py (hunk 3 above)
Comment (iszulcdeepsense) [reply racetrack_client/racetrack_client/main.py]: Good point

File: lifecycle/lifecycle/deployer/redeploy.py (hunk 2 above)
Comment (iszulcdeepsense) [reply lifecycle/lifecycle/deployer/redeploy.py:115]: Actually, I like trailing commas. This is my habit taken from Go, where it's obligatory. Sure, I can remove it in here.

File: racetrack_client/racetrack_client/main.py (hunk 4 above)
Comment (iszulcdeepsense) [reply racetrack_client/racetrack_client/main.py:81]: Right, I've noticed that too and changed it on other branch.

**Pull request merged**